flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/KafkaDynamicTableFactoryTest.java

```diff
@@ -963,6 +980,7 @@ private static KafkaDynamicSink createExpectedSink(
         tableOptions.put("properties.bootstrap.servers", "dummy");
         tableOptions.put("scan.startup.mode", "specific-offsets");
         tableOptions.put("scan.startup.specific-offsets", PROPS_SCAN_OFFSETS);
+        tableOptions.put("scan.end.specific-offsets", PROPS_SCAN_BOUNDED_OFFSETS);
```

Comment:
I don't think it is necessary to change the default settings, but it would be helpful to add a specific unit test (via mocks) to show that the bounded offsets are correctly set up.